### PR TITLE
feat: support opensearch approximate k-NN

### PR DESCRIPTION
**Pull request by baojingyu**
# Description

This PR introduces support for opensearch approximate k-NN feature, leveraging opensearch for vector retrieval and storage. It includes the integration of opensearch for approximate k-NN search and vector operations, enhancing search functionality to enable faster and more efficient retrieval of similar items.

## Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement

# How Has This Been Tested?

- [x] Unit tests were conducted to validate the functionality of opensearch approximate k-NN and vector operations integration.
- [x] The following tests cover the new features and integration: [Describe the tests here]
- [x] Test Configuration Details: [Include any relevant details for test setup]

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I ran `dev/reformat` (backend) and `cd web && npx lint-staged` (frontend) to appease the lint gods
- [x] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

File: api/config.py [hunk 1]
```diff
@@ -282,6 +282,13 @@ def __init__(self):
         self.MILVUS_SECURE = get_env('MILVUS_SECURE')
         self.MILVUS_DATABASE = get_env('MILVUS_DATABASE')
 
+        # OpenSearch 配置项
```
Comment (takatost) [inline api/config.py]: Pls make sure the comments are in English, thanks!

File: api/core/rag/datasource/vdb/opensearch/opensearch_vector.py [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+        if not values.get('host'):
+            raise ValueError("config OPENSEARCH_HOST is required")
+        if not values.get('port'):
+            raise ValueError("config OPENSEARCH_PORT is required")
+        return values
+
+    def create_ssl_context(self) -> ssl.SSLContext:
+        ssl_context = ssl.create_default_context()
+        ssl_context.check_hostname = False
+        ssl_context.verify_mode = ssl.CERT_NONE  # 禁用证书验证
```
Comment (takatost) [inline api/core/rag/datasource/vdb/opensearch/opensearch_vector.py]: Pls make sure the comments are in English, thanks!

File: api/core/rag/datasource/vdb/opensearch/opensearch_vector.py [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+    def add_texts(self, documents: list[Document], embeddings: list[list[float]], **kwargs):
+        actions = []
+        for i in range(len(documents)):
+            action = {
+                "_op_type": "index",
+                "_index": self._collection_name.lower(),
+                "_id": uuid4().hex,
+                "_source": {
+                    Field.CONTENT_KEY.value: documents[i].page_content,
+                    Field.VECTOR.value: embeddings[i],  # 确保这里传递的是数组
```
Comment (takatost) [inline api/core/rag/datasource/vdb/opensearch/opensearch_vector.py]: Same here.

**Review by takatost — DISMISSED** [review]

**Comment by JohnJyong** [comment]
pls sync the requirements to` pyproject.toml` and `peotry.lock`

**Comment by baojingyu** [comment]
> the

It has been modified as required, please review.

**Review by JohnJyong — APPROVED** [review]

**Comment by JohnJyong** [comment]
LGTM

**Pull request merged**